mcs/mcs/tuples.cs

```diff
@@ -538,10 +539,18 @@ protected override Expression DoResolve (ResolveContext rc)
 					}
 
 					var element_src = tupleLiteral == null ? new MemberAccess (instance, NamedTupleSpec.GetElementPropertyName (i)) : tupleLiteral.Elements [i].Expr;
-					targetExprs [i] = new SimpleAssign (targetExprs [i], element_src).Resolve (rc);
+
+					if(source is TupleLiteral) {
+						var temp = new LocalTemporary (element_src.Type);
+						tempExprs.Add (new SimpleAssign (temp, element_src).Resolve (rc));
+						targetExprs [i] = new SimpleAssign (targetExprs[i], temp).Resolve (rc);
+					} else {
+						targetExprs [i] = new SimpleAssign (targetExprs [i], element_src).Resolve (rc);
+					}
 				}
 
 				eclass = ExprClass.Value;
+				targetExprs.InsertRange (0, tempExprs);
```

Comment:
It should probably be better to have this as a separate field and not to add to targetExprs at all